scheduler/src/cook/kubernetes/compute_cluster.clj

```diff
@@ -43,6 +46,36 @@
             :reject-after-match-attempt true})
          node-name->available)))
 
+(defn states-to-scan
+  "Determine all states to scan by unioning task id's from expected and existing state maps. "
+  [{:keys [expected-state-map existing-state-map] :as kcc}]
+  (->>
+    (set/union (keys @expected-state-map) (keys @existing-state-map))
+    (into #{})))
+
+(defn scan-states
+  "Scan all states. Note: May block or be slow due to rate limits."
+  [kcc]
+  (log/info "Starting state scan: " )
+  ; TODO Add in rate limits; only visit non-running/running states so fast.
+  ; TODO Add in maximum-visit frequency. Only visit a state once every XX seconds.
+  (let [states-to-scan (states-to-scan kcc)
+        _ (log/info "Doing state scan. Visiting" (count states-to-scan) "states")]
+    (doseq [state states-to-scan]
+      (controller/scan-process kcc state))))
+
+(defn regular-scanner
+  "Trigger a channel that scans all states (shortly) after this function is invoked and on a regular interval."
+  [kcc interval]
+  (let [ch (async/chan (async/sliding-buffer 1))]
+    ; We scan on startup as we load the existing state in, so first scan should be scheduled for a while in the future.
+    (async/pipe (chime-ch (periodic-seq (-> interval time/from-now) interval)) ch)
+    (tools/chime-at-ch ch
+                       (fn scan-states-function []
+                         (scan-states kcc)
+                         {:error-handler (fn [e]
```

Comment:
I think the parens are off here: the function is returning `{:error-handler ...}` but it should be at the same level as the function.

Reply:
Yeah. You're right.